auth0/src/test/java/com/auth0/android/Auth0Test.java

```diff
@@ -191,4 +194,18 @@ public void shouldThrowWhenHttpUppercaseDomainUsed() {
     public void shouldThrowWhenConfigDomainIsHttp() {
         Assert.assertThrows("Invalid domain url: 'http://" + OTHER_DOMAIN + "'. Only HTTPS domain URLs are supported. If no scheme is passed, HTTPS will be used.", IllegalArgumentException.class, () -> new Auth0(CLIENT_ID, DOMAIN, "HTTP://" + OTHER_DOMAIN));
     }
+
+    @Test
+    public void shouldEnsureAuthorizeUrlIsOpen() throws NoSuchMethodException {
+        Method method = Auth0.class.getMethod("getAuthorizeUrl");
+        Assert.assertTrue(Modifier.isPublic(method.getModifiers()));
+        Assert.assertFalse(Modifier.isFinal(method.getModifiers()));
+    }
```

Comment:
This tests that the `val` is `open`, but it can be very easily broken to trigger a false positive:
 * if someone makes the `Auth0` class non-`open` in the future, it's not a compile error, and this test will still pass, but in essence inheritance is forbidden and therefore the functionality the test is verifying is broken.
 * if someone makes the `Auth0` class have no protected/public constructors, the class becomes effectively final again. Adding a private constructor like `open class Auth0 private constructor() {` will not even emit a warning, it's all good from `kotlinc`'s point of view, but the class becomes non-extensible.
 * Removing `JvmOverloads` might also have a bad effect.

So I would propose to write this test this way and let the compiler verify it:
```java
@Test
public void shouldAllowOverridingAuthorizeUrl() {
	class MyAuth extends Auth0 {
		public MyAuth() {
			super(CLIENT_ID, DOMAIN);
		}
		@Override
		public @NotNull String getAuthorizeUrl() {
			// Calling super to make sure it's fully visible as well as overridable.
			return super.getAuthorizeUrl() + "something";
		}
	}
}
```
alternatively:
```java
@Test
public void shouldAllowOverridingAuthorizeUrl() {
	// Creating an instance as well, as opposed to just `class MyAuth extends Auth0`
	// to make sure that the constructor is not doing funny business.
	new Auth0(CLIENT_ID, DOMAIN) {
		@Override
		public @NotNull String getAuthorizeUrl() {
			// Calling super to make sure it's fully visible as well as overridable.
			return super.getAuthorizeUrl() + "something";
		}
	}
}
```

Reply:
Thanks @TWiStErRob, that is a great catch. I have added this to our test suite